torch/csrc/api/src/random.cpp

```diff
@@ -0,0 +1,86 @@
+#include <torch/random.h>
+
+#include <ATen/Context.h>
+
+#include <TH/TH.h>
+#include <ATen/Context.h>
+#include <torch/types.h>
+#include <torch/csrc/utils/tensor_types.h>
+
+#ifdef USE_CUDA
+#include <THC/THCTensorRandom.h>
+#endif
```

Comment:
I'm... pretty sure you're in the non-Python bindings part of libtorch that doesn't allow you to use `USE_CUDA` in fbcode. These references will need to be indirected.

Reply:
CUDAHooks is probably the right place for the indirection to live, for now.